awx/api/views.py

```diff
@@ -145,6 +145,16 @@ def destroy(self, request, *args, **kwargs):
         # Still allow deletion of new status, because these can be manually created
         if obj.status in ACTIVE_STATES and obj.status != 'new':
             raise PermissionDenied(detail=_("Cannot delete running job resource."))
+        elif not obj.events_processed:
+            # Prohibit deletion if job events are still coming in
+            if obj.finished and now() < obj.finished + dateutil.relativedelta.relativedelta(minutes=1):
+                # less than 1 minute has passed since job finished and events are not in
+                return Response({"error": _("Job has not finished processing events.")},
+                                status=status.HTTP_400_BAD_REQUEST)
+            else:
+                # if it has been > 1 minute, events are probably lost
```

Comment:
Should we _only_ warn in this way if `obj.finished`?  What if you deleted a job that wasn't finished yet, but was in some other state (like `pending`)?

Reply:
Yes. In `tasks.py`, we have a line

```python
instance = self.update_model(pk, status=status, result_traceback=tb,
```

This is the canonical finalization save for a job. This should be the first and only time that a job's status changes from an active state to a finalized state.

This new field was piggybacked onto this save.

In the `.save` method for unified jobs, it also tacks on `finished=now()` if the status changed to a finalized state. That means that the finished time gets saved at the same time that the event count does.

I'm worried about preventing false-positives here, and not false-negatives. If the job is still running, then its deletion should already be blocked by other logic. If that logic fails, then I don't want _this_ code to throw a traceback and confuse the issue.

Reply:
👍